packages/client/wallets/smart-wallet/src/blockchain/wallets/SendTransactionService.ts

```diff
@@ -161,10 +163,10 @@ export class SendTransactionService {
         stage: SendTransactionFailureStage
     ) {
         try {
-            await this.publicClient.simulateContract({
+            await this.client.public.simulateContract({
                 ...request,
-                account: request.address,
```

Comment:
This is line w/ the bug. `request.address` is the contract address, and not the address for the wallet.